src/amqp10/properties.ts

```diff
@@ -0,0 +1,92 @@
+import { MessageProperties } from "../producer"
+import { DataReader } from "../responses/raw_response"
+
+export class Properties {
+  public static Parse(dataResponse: DataReader, fields: number): MessageProperties {
+    let messageId = ""
+    let to = ""
+    let subject = ""
+    let replyTo = ""
+    let correlationId = ""
+    let contentType = ""
+    let contentEncoding = ""
+    let userId = Buffer.from("")
+    let absoluteExpiryTime = new Date()
+    let creationTime = new Date()
+    let groupId = ""
+    let groupSequence = 0
+    let replyToGroupId = ""
+
+    for (let index = 0; index < fields; index++) {
```

Comment:
bonus point if you can replace this for with a reduce

Reply:
Same as the reply above

Reply:
range(fields).reduce(..., {})
@albertobarrila can help you ...

Reply:
Pay attention we don't have lodash so we dont' have `¶ange` but If remember correctly we have implemented it in some test ...